ambry-cloud/src/main/java/com.github.ambry.cloud/VcrReplicationManager.java

```diff
@@ -169,11 +171,9 @@ void addPartition(PartitionId partitionId) throws ReplicationException {
     }
     ReplicaId cloudReplica =
         new CloudReplica(cloudConfig, partitionId, virtualReplicatorCluster.getCurrentDataNodeId());
-    Store cloudStore = new CloudBlobStore(properties, partitionId, cloudDestination, clusterMap, vcrMetrics);
-    try {
-      cloudStore.start();
-    } catch (StoreException e) {
-      throw new ReplicationException("Can't start CloudStore " + cloudStore, e);
+    if (!storeManager.addBlobStore(cloudReplica)) {
```

Comment:
We discussed a little bit about similar logic of `addPartition` in `VcrReplicationManager` and `addReplica` in regular `ReplicationManager`. If we try to make it consistent and put single abstract method in base class, I feel like `addReplica` might be a more appropriate method name considering nature of `VCR` is actually a virtual replica. Helix controller assigns a replica of certain partition to this VCR node. In this `addPartition` method, the partitionId is used to create cloudReplica. We can move this out and pass in the created cloudReplica. Correct me if I am wrong.